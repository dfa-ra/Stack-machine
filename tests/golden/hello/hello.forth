_data_
"Hello world" VAR str
0x84 VAR output_addr

_func_
: PRINT
    !iter
    LOOP
        1 !tmp 0 != 0
        WHILE
            3 iter - a! @
            0xFF000000 and
            dup dup !tmp
            0 = IF
            ELSE
                output_addr a! !
                iter 1 + !iter
            THEN
        REPEAT
;

_text_
&str PRINT
HALT
