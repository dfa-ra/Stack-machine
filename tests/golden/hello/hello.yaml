name: hello
limit: 500
memory_size: 200
input_streams:
  0x80: []
  0x84: []
reports:
  - name: binary
    slice: mem
    view: |
      {instruction}
  - name: Step-by-step
    slice: all
    view: |
      tick: {cpu:tick} || microcode - {microcode} || A: {cpu:A:hex} || B: {cpu:B:hex} 
      data stack: {stack:data}
  - name: Result
    slice: last
    view: |
      numio[0x80]: {io:in:dec}
      numio[0x84]: {io:out:dec}
      tick_count = {cpu:tick}
      {condition:[io:out:dec]=[72, 101, 108, 108, 111, 32, 119, 111, 114, 108, 100]}
      {condition:[io:out:str]=[Hello world]}